dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -347,6 +353,16 @@ public void close() {
                     if (hasSubstantialRemainingData(response)) {
                         ExecRuntime runtime = HttpClientExecRuntimeAttributeInterceptor.get(context);
                         if (runtime != null) {
+                            // Attempt to set the smallest possible socket timeout before closing the connection.
+                            // In some degenerate cases, remote servers may not close connections, causing the client
+                            // to consume network resources and time in SSLSocketInputRecord.deplete.
+                            ConnectionEndpoint maybeEndpoint = ConnectionEndpointAccess.getConnectionEndpoint(runtime);
+                            assert maybeEndpoint != null || !runtime.isEndpointConnected()
+                                    : "Expected ConnectionEndpointAccess.getConnectionEndpoint "
+                                            + "to extract a ConnectionEndpoint";
+                            if (maybeEndpoint != null) {
+                                maybeEndpoint.setSocketTimeout(Timeout.ONE_MILLISECOND);
```

Comment:
Should we set this to 0 so that the deplete `tryToRead` arg is false
https://github.com/openjdk/jdk17u/blob/2fe42855c48c49b515b97312ce64a5a8ef3af407/src/java.base/share/classes/sun/security/ssl/SSLSocketImpl.java#L1221

Reply:
Yeah, makes sense so we should expect the skip to timeout after 1ms with no bytes